examples/chip-tool-darwin/commands/clusters/ClusterCommandBridge.h

```diff
@@ -0,0 +1,129 @@
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#pragma once
+
+#import <CHIP/CHIP.h>
+#import <CHIP/CHIPDevice_Internal.h>
+#include <lib/support/UnitTestUtils.h>
+
+#include "ModelCommandBridge.h"
+
+class ClusterCommand : public ModelCommand {
+public:
+    ClusterCommand()
+        : ModelCommand("command-by-id")
+    {
+        AddArgument("cluster-id", 0, UINT32_MAX, &mClusterId);
+        AddArgument("command-id", 0, UINT32_MAX, &mCommandId);
+        AddArgument("payload", &mPayload);
+        AddArgument("timedInteractionTimeoutMs", 0, UINT16_MAX, &mTimedInteractionTimeoutMs);
+        AddArgument("repeat-count", 1, UINT16_MAX, &mRepeatCount);
+        AddArgument("repeat-delay-ms", 0, UINT16_MAX, &mRepeatDelayInMs);
+        ModelCommand::AddArguments();
+    }
+
+    ClusterCommand(chip::ClusterId clusterId)
+        : ModelCommand("command-by-id")
+        , mClusterId(clusterId)
+    {
+        AddArgument("command-id", 0, UINT32_MAX, &mCommandId);
+        AddArgument("payload", &mPayload);
+        AddArgument("timedInteractionTimeoutMs", 0, UINT16_MAX, &mTimedInteractionTimeoutMs);
+        AddArgument("repeat-count", 1, UINT16_MAX, &mRepeatCount);
+        AddArgument("repeat-delay-ms", 0, UINT16_MAX, &mRepeatDelayInMs);
+        ModelCommand::AddArguments();
+    }
+
+    ClusterCommand(const char * _Nonnull commandName)
+        : ModelCommand(commandName)
+    {
+        AddArgument("timedInteractionTimeoutMs", 0, UINT16_MAX, &mTimedInteractionTimeoutMs);
+        AddArgument("repeat-count", 1, UINT16_MAX, &mRepeatCount);
+        AddArgument("repeat-delay-ms", 0, UINT16_MAX, &mRepeatDelayInMs);
+    }
+
+    ~ClusterCommand() {}
+
+    CHIP_ERROR SendCommand(CHIPDevice * _Nullable device, chip::EndpointId endpointId) override
+    {
+        chip::TLV::TLVWriter writer;
+        chip::TLV::TLVReader reader;
+
+        mData = static_cast<uint8_t *>(chip::Platform::MemoryCalloc(sizeof(uint8_t), mDataMaxLen));
+        VerifyOrReturnError(mData != nullptr, CHIP_ERROR_NO_MEMORY);
+
+        writer.Init(mData, mDataMaxLen);
+
+        mPayload.Encode(writer, chip::TLV::AnonymousTag());
+        reader.Init(mData, writer.GetLengthWritten());
+        reader.Next();
+
+        id commandFields = NSObjectFromCHIPTLV(&reader);
+
+        return ClusterCommand::SendCommand(device, endpointId, mClusterId, mCommandId, commandFields);
+    }
+
+    CHIP_ERROR SendCommand(CHIPDevice * _Nullable device, chip::EndpointId endpointId, chip::ClusterId clusterId,
+        chip::CommandId commandId, id _Nonnull commandFields)
+    {
+        uint16_t repeatCount = mRepeatCount.ValueOr(1);
+        NSDictionary * commandFieldDict = @{@"type" : @"Structure", @"value" : @[]};
+        dispatch_queue_t callbackQueue = dispatch_queue_create("com.chip.command", DISPATCH_QUEUE_SERIAL);
+        if (commandFields != nullptr) {
```

Comment:
Since there was not documentation on how to use I looked up a usage and on this [line](https://github.com/project-chip/connectedhomeip/blob/master/src/darwin/Framework/CHIP/CHIPDevice.mm#L545) it used nullptr, but now that you point it out, the return is nil. I will update, thank you.

Reply:
Updated.